health/notifications/alarm-notify.sh.in

```diff
@@ -481,197 +463,176 @@ filter_recipient_by_criticality() {
 [ -z "${PUSHBULLET_ACCESS_TOKEN}" ] && SEND_PUSHBULLET="NO"
 
 # check twilio
-[ -z "${TWILIO_ACCOUNT_TOKEN}" -o -z "${TWILIO_ACCOUNT_SID}" -o -z "${TWILIO_NUMBER}" ] && SEND_TWILIO="NO"
+{ [ -z "${TWILIO_ACCOUNT_TOKEN}" ] || [ -z "${TWILIO_ACCOUNT_SID}" ] || [ -z "${TWILIO_NUMBER}" ]; } && SEND_TWILIO="NO"
 
 # check hipchat
 [ -z "${HIPCHAT_AUTH_TOKEN}" ] && SEND_HIPCHAT="NO"
 
 # check messagebird
-[ -z "${MESSAGEBIRD_ACCESS_KEY}" -o -z "${MESSAGEBIRD_NUMBER}" ] && SEND_MESSAGEBIRD="NO"
+{ [ -z "${MESSAGEBIRD_ACCESS_KEY}" ] || [ -z "${MESSAGEBIRD_NUMBER}" ]; } && SEND_MESSAGEBIRD="NO"
 
 # check kavenegar
-[ -z "${KAVENEGAR_API_KEY}" -o -z "${KAVENEGAR_SENDER}" ] && SEND_KAVENEGAR="NO"
+{ [ -z "${KAVENEGAR_API_KEY}" ] || [ -z "${KAVENEGAR_SENDER}" ]; } && SEND_KAVENEGAR="NO"
 
 # check telegram
 [ -z "${TELEGRAM_BOT_TOKEN}" ] && SEND_TELEGRAM="NO"
 
 # check kafka
-[ -z "${KAFKA_URL}" -o -z "${KAFKA_SENDER_IP}" ] && SEND_KAFKA="NO"
+{ [ -z "${KAFKA_URL}" ] || [ -z "${KAFKA_SENDER_IP}" ]; } && SEND_KAFKA="NO"
 
 # check irc
 [ -z "${IRC_NETWORK}" ] && SEND_IRC="NO"
 
 # check fleep
-[ -z "${FLEEP_SERVER}" -o -z "${FLEEP_SENDER}" ] && SEND_FLEEP="NO"
+#shellcheck disable=SC2153
+{ [ -z "${FLEEP_SERVER}" ] || [ -z "${FLEEP_SENDER}" ]; } && SEND_FLEEP="NO"
 
 # if we need curl, check for the curl command
-if [ \( \
-           "${SEND_PUSHOVER}"    = "YES" \
-        -o "${SEND_SLACK}"       = "YES" \
-        -o "${SEND_ROCKETCHAT}"  = "YES" \
-        -o "${SEND_ALERTA}"      = "YES" \
-        -o "${SEND_PD}"          = "YES" \
-        -o "${SEND_FLOCK}"       = "YES" \
-        -o "${SEND_DISCORD}"     = "YES" \
-        -o "${SEND_HIPCHAT}"     = "YES" \
-        -o "${SEND_TWILIO}"      = "YES" \
-        -o "${SEND_MESSAGEBIRD}" = "YES" \
-        -o "${SEND_KAVENEGAR}"   = "YES" \
-        -o "${SEND_TELEGRAM}"    = "YES" \
-        -o "${SEND_PUSHBULLET}"  = "YES" \
-        -o "${SEND_KAFKA}"       = "YES" \
-        -o "${SEND_FLEEP}"       = "YES" \
-        -o "${SEND_PROWL}"       = "YES" \
-        -o "${SEND_CUSTOM}"      = "YES" \
-        -o "${SEND_MSTEAM}"        = "YES" \
-    \) -a -z "${curl}" ]
-    then
-    curl="$(which curl 2>/dev/null || command -v curl 2>/dev/null)"
-    if [ -z "${curl}" ]
-        then
-        error "Cannot find curl command in the system path. Disabling all curl based notifications."
-        SEND_PUSHOVER="NO"
-        SEND_PUSHBULLET="NO"
-        SEND_TELEGRAM="NO"
-        SEND_SLACK="NO"
-        SEND_MSTEAM="NO"
-        SEND_ROCKETCHAT="NO"
-        SEND_ALERTA="NO"
-        SEND_PD="NO"
-        SEND_FLOCK="NO"
-        SEND_DISCORD="NO"
-        SEND_TWILIO="NO"
-        SEND_HIPCHAT="NO"
-        SEND_MESSAGEBIRD="NO"
-        SEND_KAVENEGAR="NO"
-        SEND_KAFKA="NO"
-        SEND_FLEEP="NO"
-        SEND_PROWL="NO"
-        SEND_CUSTOM="NO"
-    fi
+if [ -z "${curl}" ]; then
+	curl="$(command -v curl 2>/dev/null)"
+fi
+if [ -z "${curl}" ]; then
```

Comment:
It's not an error if I never wanted to send one of those notifications in the first place. We either need the check on whether any of the curl-based notifications are enabled, or we can remove this part completely and have the check executed in function `docurl` itself.

Reply:
It is not erroring anywhere. This code does similar logic as it was doing before refactoring.

This PR is about linting and readability, not about code logic.

Reply:
The logic is being changed, because the check on whether any such notification is enabled was removed. Right after the -z curl there is an `error` line.

Reply:
fixed